### PR TITLE
Facebook Auth Plugin

### PR DESCRIPTION
**Pull request by wyattjoh**
## What does this PR do?

- Extracts server side facebook auth into plugin, enabled by default

## How do I test this PR?

- Try with facebook auth plugin enabled
- Notice it works :)
- Try without facebook auth plugin enabled
- Notice that facebook auth does not work :)

### CODE REVIEW

File: plugins/coral-plugin-facebook-auth/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+This plugin looks for the following configuration from the environment:
+
+- `TALK_FACEBOOK_APP_ID` (*required*) - the Facebook app id for your Facebook
+Login enabled app.
+- `TALK_FACEBOOK_APP_SECRET` (*required*) - the Facebook app secret for your
+Facebook Login enabled app.
+
+### License
+
+    Copyright 2016 Mozilla Foundation
```
Comment (riley) [inline plugins/coral-plugin-facebook-auth/README.md]: is our stuff copyrighted under mozilla or the Coral Project as a separate identity? @losowsky

File: plugins/coral-plugin-facebook-auth/server/router.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module.exports = (router) => {
+
+  const {passport, HandleAuthPopupCallback} = require('services/passport');
+
+  /**
+   * Facebook auth endpoint, this will redirect the user immediatly to facebook
+   * for authorization.
+   */
+  router.get('/api/v1/auth/facebook', passport.authenticate('facebook', {display: 'popup', authType: 'rerequest', scope: ['public_profile']}));
```
Comment (riley) [inline plugins/coral-plugin-facebook-auth/server/router.js:9]: is there a way to pass config stuff into `plugins.json` so they don't have to edit code if they want to change the scope or something?

**Review by riley — CHANGES_REQUESTED** [review]

File: plugins/coral-plugin-facebook-auth/README.md (hunk 1 above)
Comment (losowsky) [reply plugins/coral-plugin-facebook-auth/README.md]: Apache 2 license, IP only is Mozilla's

Comment (riley) [reply plugins/coral-plugin-facebook-auth/README.md]: 👍

File: plugins/coral-plugin-facebook-auth/server/router.js (hunk 2 above)
Comment (wyattjoh) [reply plugins/coral-plugin-facebook-auth/server/router.js:9]: No, they should have to edit the code to adjust behaviour.

**Review by riley — APPROVED** [review]

**Pull request merged**